### PR TITLE
feat: `invoke` - Adding option allowing insecure connections over SSL

### PR DESCRIPTION
**Pull request by zroubalik**
Signed-off-by: Zbynek Roubalik <zroubalik@gmail.com>

# Changes

- :gift: `invoke` - Adding option allowing insecure connections over SSL

/kind enhancement

**Release Note**

```release-note
`invoke` - Adding option allowing insecure connections over SSL
```

### CODE REVIEW

File: cmd/invoke.go [hunk 1]
```diff
@@ -375,5 +382,17 @@ func (c invokeConfig) prompt(client *fn.Client) (invokeConfig, error) {
 		return c, err
 	}
 
+	qs = []*survey.Question{
+		{
+			Name: "Insecure",
+			Prompt: &survey.Confirm{
+				Message: "Allow insecure server connections when using SSL",
+				Default: false,
```
Comment (lkingland) [inline cmd/invoke.go:390]: The default could be set to the current value of `cfg.Insecure` rather than `false` such that a call to `--confirm` which also includes `--insecure` will set the confirm prompt appropriately.
```shell
func invoke --insecure --confirm
```
... would then prompt but with the default set to `true`

**Review by lkingland — APPROVED** [review]
Nice!  I might just consider changing that default value in the prompt to be dynamic
/lgtm
/hold for others

File: cmd/invoke.go (hunk 1 above)
Comment (zroubalik) [reply cmd/invoke.go:390]: Right!

Comment (zroubalik) [reply cmd/invoke.go]: Fixed

File: docs/reference/func_deploy.md [hunk 2]
```diff
@@ -95,7 +95,7 @@ func deploy
   -p, --path string             Path to the project directory (Env: $FUNC_PATH) (default ".")
       --platform string         Target platform to build (e.g. linux/amd64).
   -u, --push                    Push the function image to registry before deploying (Env: $FUNC_PUSH) (default true)
-  -r, --registry string         Registry + namespace part of the image to build, ex 'ghcr.io/myuser'.  The full image name is automatically determined based on the local directory name. If not provided the registry will be taken from func.yaml (Env: $FUNC_REGISTRY)
+  -r, --registry string         Registry + namespace part of the image to build, ex 'ghcr.io/myuser'.  The full image name is automatically determined. (Env: $FUNC_REGISTRY) (default "image-registry.openshift-image-registry.svc:5000/demo")
```
Comment (lance) [inline docs/reference/func_deploy.md:98]: This change seems unrelated to the overall PR - and I don't think that the default in upstream is the openshift image registry.

**Review by lance — COMMENTED** [review]
Overall great contribution - thanks! I'm just not sure about the change to func_deploy.md

File: docs/reference/func_deploy.md (hunk 2 above)
Comment (zroubalik) [reply docs/reference/func_deploy.md:98]: I have no idea how does this actually appeared here.

Comment (zroubalik) [reply docs/reference/func_deploy.md]: Good catch, thanks. Fixed.

**Comment by matejvasek** [comment]
Would it make sense to have this as top level flag? It could be for instance useful for users with self-signed private OCI registry.

**Comment by matejvasek** [comment]
@jrangelramos would you find useful to have the flag global for testing purposes?

**Comment by matejvasek** [comment]
Thinking of it more deeply: it wouldn't be easy to implement this for `push` operation,
because sometimes push is done directly by `docker` daemon, and there is no easy way to force daemon do it.

**Comment by jrangelramos** [comment]
@matejvasek I think ti would be useful for deploy command for sure, specially if the user options could be forwarded to on-cluster builds deploy step as well

File: docs/reference/func.md [hunk 3]
```diff
@@ -4,7 +4,7 @@ Serverless functions
 
 ### Synopsis
 
-Serverless functions v0.0.0-source-2022-09-12T14:54:32-06:00
+Serverless functions v0.0.0-source-2022-09-19T12:01:46&#43;02:00
```
Comment (matejvasek) [inline docs/reference/func.md:7]: This probably doesn't have to be part of the PR?

File: docs/reference/func_deploy.md [hunk 4]
```diff
@@ -12,7 +12,7 @@ SYNOPSIS
 	func deploy [-R|--remote] [-r|--registry] [-i|--image] [-n|--namespace]
 	             [-e|env] [-g|--git-url] [-t|git-branch] [-d|--git-dir]
 	             [-b|--build] [--builder] [--builder-image] [-p|--push]
-	             [-c|--confirm] [-v|--verbose]
+	             [--platform] [-c|--confirm] [-v|--verbose]
```
[the remark this answers is not in the document]
Comment (matejvasek) [reply docs/reference/func_deploy.md:15]: Should it be part of this PR?

**Comment by matejvasek** [comment]
> if the user options could be forwarded to on-cluster builds deploy step as well. For sure that will save some user of some headaches.. :)

I am afraid that would be out of scope. We cannot easily force `docker daemon` or `k8s cluster` to trust connections.

**Comment by matejvasek** [comment]
Looks good, maybe with exception of the two `*.md` file changes.

File: docs/reference/func_deploy.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (zroubalik) [reply docs/reference/func_deploy.md:15]: The `--platform` was probably missing in the generated doc. This change is just making the docs correct, or not?

[the remark this answers is not in the document]
Comment (lkingland) [reply docs/reference/func_deploy.md:15]: This seems a reasonable cleanup IMHO.  There was a missing flag name in the help text synopsis.  This replicates the fixing of that omission to the generated help text.  Seems fine?

**Comment by zroubalik** [comment]
I have removed the docs fix from this PR and created another one. @matejvasek PTAL
